src/em/prov/em_provisioning.cpp

```diff
@@ -543,6 +545,121 @@ int em_provisioning_t::handle_proxy_encap_dpp(uint8_t *buff, unsigned int len)
 	return 0;
 }
 
+cJSON *em_provisioning_t::create_enrollee_bsta_list(ec_connection_context_t *conn_ctx)
+{
+    dm_easy_mesh_t *dm = get_data_model();
+    if (dm == nullptr) {
+        printf("%s:%d: Could not get data model handle!\n", __func__, __LINE__);
+        return nullptr;
+    }
+
+    std::string channelList;
+    cJSON *bsta_list_obj = cJSON_CreateObject();
+    if (!bsta_list_obj) {
+        printf("%s:%d: Failed to allocate for bSTAList object!\n", __func__, __LINE__);
+        return nullptr;
+    }
+
+    cJSON *b_sta_list_arr = cJSON_CreateArray();
+    if (!b_sta_list_arr) {
+        printf("%s:%d: Could not allocate for bSTAList array!\n", __func__, __LINE__);
+        cJSON_Delete(bsta_list_obj);
+        return nullptr;
+    }
+
+    if (!cJSON_AddStringToObject(bsta_list_obj, "netRole", "mapBackhaulSta")) {
+        printf("%s:%d: Could not add netRole to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(b_sta_list_arr);
+        return nullptr;
+    }
+
+    // XXX: TODO: akm is hard-coded. Should come from em_akm_suite_info_t or equivalent, but
+    // not currently populated anywhere in the data model.
+    std::string akm = "psk";
+    if (!cJSON_AddStringToObject(bsta_list_obj, "akm", akm.c_str())) {
+        printf("%s:%d: Could not add AKM to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(b_sta_list_arr);
+        return nullptr;
+    }
+
+    if (!cJSON_AddNumberToObject(bsta_list_obj, "bSTA_Maximum_Links", 1)) {
+        printf("%s:%d: Could not add bSTA_Maximum_Links to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(b_sta_list_arr);
+        return nullptr;
+    }
+
+    if (!cJSON_AddItemToArray(b_sta_list_arr, bsta_list_obj)) {
+        printf("%s:%d: Could not add bSTAList object to bSTAList array!\n", __func__, __LINE__);
+        cJSON_Delete(b_sta_list_arr);
+        return nullptr;
+    }
+
+    cJSON *radio_list_arr = cJSON_AddArrayToObject(bsta_list_obj, "RadioList");
+    if (!radio_list_arr) {
+        printf("%s:%d: Could not add RadioList array to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(bsta_list_obj);
+        return nullptr;
+    }
+
+    for (unsigned int i = 0; i < dm->get_num_radios(); i++) {
+        const dm_radio_t *radio = dm->get_radio(i);
+        if (!radio)
+            continue;
+
+        cJSON *radioListObj = cJSON_CreateObject();
+        if (!radioListObj) {
+            printf("%s:%d: Failed to create RadioList object!\n", __func__, __LINE__);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
+
+        if (!cJSON_AddStringToObject(
+                radioListObj, "RUID",
+                util::mac_to_string(radio->m_radio_info.id.ruid, "").c_str())) {
+            printf("%s:%d: Could not add RUID to RadioList object!\n", __func__, __LINE__);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
```

Comment:
Based on the example previously given this seems *fine* but its weird that it doesn't free the `radioListObj`.
I can't tell if that's an oversight in the example or intentional. Either way, you do it later in the code so it might be a good idea to just do it here.

```suggestion
        if (!cJSON_AddStringToObject(
                radioListObj, "RUID",
                util::mac_to_string(radio->m_radio_info.id.ruid, "").c_str())) {
            printf("%s:%d: Could not add RUID to RadioList object!\n", __func__, __LINE__);
            cJSON_Delete(bsta_list_obj);
            cJSON_Delete(radioListObj);
            return nullptr;
        }
```